FashionConnect_ui/src/App.jsx

```diff
@@ -30,18 +30,60 @@ function App() {
     // Save the user data to storage whenever the user state changes
     localStorage.setItem('user', JSON.stringify(user));
   }, [user]);
- 
+
+  const [collections, setCollections] = useState([]);
+  const [products, setProducts] = useState([]);
+  const userId = user.id;
+
+  useEffect(() => {
+    const fetchCollections = async () => {
+      try {
+        const user_collections = await fetch(`http://localhost:3000/collections/${userId}`);
+        const data = await user_collections.json();
+        setCollections(data);
+      } catch (error) {
+        console.error('Error fetching collections:', error);
+        throw error;
+      }
+    };
+
+    fetchCollections();
+  }, [userId]);
+
+  useEffect(() => {
+    const fetchProducts = async () => {
+      const productPromises = collections.flatMap((collection) =>
+        collection.ProductID.map((productId) => fetchProductDetails(productId))
+      );
+      const finalProducts = await Promise.all(productPromises);
+      setProducts(finalProducts);
+    };
+
+    fetchProducts();
+  }, [collections]);
+
+  const fetchProductDetails = async (productId) => {
+    try {
+      const diff_products = await fetch(`http://localhost:3000/products/${productId}`);
+      const product = await diff_products.json();
+      return product;
+    } catch (error) {
+      console.error('Error fetching product details:', error);
```

Comment:
Let's clean up all the console error logs please.

Reply:
oh okay.
I will remove them as from now